src/Raven.Client/Util/JavascriptConversionExtensions.cs

```diff
@@ -2671,6 +2671,194 @@ public override void ConvertToJavascript(JavascriptConversionContext context)
             }
         }
 
+        /// <summary>
+        /// This is a copy of Lambda2Js.EnumConversionExtension with modifications to always treat dictionary enum keys as strings (isDictionaryKey).
```

Comment:
> Did you make any additional changes here? 

Yes, the changes are that we need to know if the current item is a dictionary key (`isDictionaryKey`) and in that case always use the string value because that's the way store handles it. Even when you set SaveEnumsAsIntegers, dictionary keys are still strings.

There is also a lot of unused code in that file that is never used but I did not know if it's best to keep the code as close to the original (as now) of if I should "clean" it up to only use the required flows.

Reply:
I looks that I didn't read the entire comment where you said:
> ... with modifications to always treat dictionary enum keys as strings (isDictionaryKey)

That's okay then. Can you please also reference the permalink to the original source code in Lambda2Js repo?

Reply:
You were right. I could remove the entire duplication and simplify the solution. Since this is only used for patching **paths** we can always use the string representation regardless of `SaveEnumsAsIntegers` or not.